x-pack/legacy/plugins/security/server/lib/auth_redirect.js

```diff
@@ -15,6 +15,29 @@ import { wrapError } from './errors';
  * request that needs to be authenticated.
  */
 export function authenticateFactory(server) {
+
+  // Sometimes authentication subsystem may require some special headers to be sent back to
+  // the client with the requested payload (e.g. for mutual client-server authentication).
+  const authResponseHeadersMap = new WeakMap();
```

Comment:
__note:__ that's the main change. And I'm not a big fan of it, but we won't need it in 7.4 when https://github.com/elastic/kibana/pull/41689 is implemented.